game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/battlegroup/BombardmentBattleGroups.java

```diff
@@ -0,0 +1,63 @@
+package games.strategy.triplea.delegate.battle.steps.fire.battlegroup;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.NAVAL_BOMBARD;
+
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.List;
+import java.util.function.Predicate;
+import lombok.experimental.UtilityClass;
+
+/** Constructs battle groups for the bombardment phase of a battle */
+@UtilityClass
+class BombardmentBattleGroups {
```

Comment:
As an idea, instead of one utility class per unit ability, perhaps we can create an interface to capture the key characteristics of a unit ability. Then battle group would be more generic and would only need a set of unit ability interfaces that it could then use to do any calculation.